### ISSUE TITLE
MO2 hangs before launch.

### ISSUE THREAD

**Issue by tetoNidan** [sonic2kk/steamtinkerlaunch#715]
## System Information
- steamtinkerlaunch-v12.0
- Manjaro
- AUR - pamac build steamtinkerlaunch

## Issue Description

Latest STL hangs trying to open MO2. Not using steam beta if that makes any difference.
So I thought I'd try out the new STL after trying steam beta. Steams new UI seems nice but cant get controller to work after launch of MO2. With and without firejail so I left beta again, installed the latest STL from the AUR ans STL opens up just fine with the splash screen but MO2 never opens. 

Sorry it takes me a long time to respond as I live in the sticks and have spotty internet connection at best. Im going to reinstall the old stl-git and hope I can play but will reinstall STL if there is something I can do to provide more information or suggested troubleshooting. I did start steam in the terminal and there was allot of output. Way to much to post. Most of it was these lines below. This output is without firejail sandboxing steam.

Any way thanks STL team!

I tried to revert back to the STL-git and get the same issue but I did notice this in the output of steam in the terminal that I forgot to add previously:

mv: cannot move '/home/shop/.local/share/applications/ModOrganizer-steamtinkerlaunch-dl.desktop' to '/home/shop/.local/share/applications/ModOrganizer-steamtinkerlaunch-dl.desktop-old': Read-only file system
gamemodeauto:

-----

Child received signal 2, shutting down the sandbox...
gamemodeauto: 
pid 28494 != 3620, skipping destruction (fork without exec?)
/home/shop/.local/share/Steam/compatibilitytools.d/SteamTinkerLaunch/steamtinkerlaunch: line 6072: echo: write error: Broken pipe
gamemodeauto: 
pid 28492 != 3620, skipping destruction (fork without exec?)
Terminated
gamemodeauto: 
pid 28498 != 3620, skipping destruction (fork without exec?)
gamemodeauto: 
pid 28496 != 3620, skipping destruction (fork without exec?

## Logs

Mon Jan  9 07:15:21 PM MST 2023 INFO - setflatpak - started SteamTinkerLaunch from /usr/bin/steamtinkerlaunch
Mon Jan  9 07:15:21 PM MST 2023 INFO - initAID - Set AID to PLACEHOLDERAID '31337'
Mon Jan  9 07:15:21 PM MST 2023 INFO - loadLanguage - First load the default language 'english' to make sure all variables are filled
Mon Jan  9 07:15:21 PM MST 2023 INFO - loadLangFile - Language from command line is 'english'
Mon Jan  9 07:15:21 PM MST 2023 INFO - loadLangFile - SYSTEMSTLCFGDIR is '/usr/share/steamtinkerlaunch'
Mon Jan  9 07:15:21 PM MST 2023 INFO - loadLangFile - Command line language 'english' is no file - trying to find its absolute path
Mon Jan  9 07:15:21 PM MST 2023 INFO - loadLangFile - Loading found system wide /usr/share/steamtinkerlaunch/lang//english.txt
Mon Jan  9 07:15:21 PM MST 2023 INFO - loadLanguage - Loading STLLANG from '/home/shop/.config/steamtinkerlaunch/global.conf'
Mon Jan  9 07:15:21 PM MST 2023 INFO - loadLanguage - STLLANG from '/home/shop/.config/steamtinkerlaunch/global.conf' is 'english'
Mon Jan  9 07:15:21 PM MST 2023 INFO - setLocalInstall - Looks like we don't have a local non-root install
Mon Jan  9 07:15:21 PM MST 2023 INFO - updateConfigEntry - Option 'GLOBALCOLLECTIONDIR' is updated with the new value '/usr/share/steamtinkerlaunch/collections' in config 'global.conf'
Mon Jan  9 07:15:21 PM MST 2023 INFO - getCurrentCommandline - No Proton in command line arguments '--version'
Mon Jan  9 07:15:21 PM MST 2023 INFO - saveOrgVars - Storing some original variables to restore them later
Mon Jan  9 07:15:21 PM MST 2023 INFO - emptyVars - Emptied some original variables as they slowdown several system calls when started from steam
Mon Jan  9 07:15:21 PM MST 2023 INFO - emptyVars - Set $PATH to '/home/shop/.local/bin:/usr/local/sbin:/usr/local/bin:/usr/bin:/usr/lib/jvm/default/bin:/usr/bin/site_perl:/usr/bin/vendor_perl:/usr/bin/core_perl:/var/lib/snapd/snap/bin:'
Mon Jan  9 07:15:21 PM MST 2023 START - ######### Initializing Game Launch 31337 using SteamTinkerLaunch v12.0 #########
Mon Jan  9 07:15:21 PM MST 2023 INFO - main - Starting with a clean log
Mon Jan  9 07:15:21 PM MST 2023 INFO - main - Start creating default configs
Mon Jan  9 07:15:21 PM MST 2023 INFO - createDefaultCfgs - START
Mon Jan  9 07:15:21 PM MST 2023 INFO - loadLanguage - First load the default language 'english' to make sure all variables are filled
Mon Jan  9 07:15:21 PM MST 2023 INFO - loadLangFile - Language from command line is 'english'
Mon Jan  9 07:15:21 PM MST 2023 INFO - loadLangFile - GLOBALSTLLANGDIR is '/usr/share/steamtinkerlaunch/lang'
Mon Jan  9 07:15:21 PM MST 2023 INFO - loadLangFile - Command line language 'english' is no file - trying to find its absolute path
Mon Jan  9 07:15:21 PM MST 2023 INFO - loadLangFile - Loading found system wide /usr/share/steamtinkerlaunch/lang/english.txt
Mon Jan  9 07:15:21 PM MST 2023 INFO - loadLanguage - Loading STLLANG from '/home/shop/.config/steamtinkerlaunch/global.conf'
Mon Jan  9 07:15:21 PM MST 2023 INFO - loadLanguage - STLLANG from '/home/shop/.config/steamtinkerlaunch/global.conf' is 'english'
Mon Jan  9 07:15:21 PM MST 2023 INFO - createDefaultCfgs - setSteamPaths:
Mon Jan  9 07:15:21 PM MST 2023 INFO - setSteamPath - Set 'SROOT' to '/home/shop/.local/share/Steam'
Mon Jan  9 07:15:21 PM MST 2023 INFO - setSteamPath - Set 'SUSDA' to '/home/shop/.local/share/Steam/userdata'
Mon Jan  9 07:15:21 PM MST 2023 INFO - setSteamPath - Set 'DEFSTEAMAPPS' to '/home/shop/.local/share/Steam/steamapps'
Mon Jan  9 07:15:21 PM MST 2023 INFO - setSteamPath - Set 'DEFSTEAMAPPSCOMMON' to '/home/shop/.local/share/Steam/steamapps/common'
Mon Jan  9 07:15:21 PM MST 2023 INFO - setSteamPath - Set 'CFGVDF' to '/home/shop/.local/share/Steam/config/config.vdf'
Mon Jan  9 07:15:21 PM MST 2023 INFO - setSteamPath - Set 'LFVDF' to '/home/shop/.local/share/Steam/steamapps/libraryfolders.vdf'
Mon Jan  9 07:15:21 PM MST 2023 INFO - setSteamPath - Set 'FAIVDF' to '/home/shop/.local/share/Steam/appcache/appinfo.vdf'
Mon Jan  9 07:15:21 PM MST 2023 INFO - setSteamPath - Set 'PIVDF' to '/home/shop/.local/share/Steam/appcache/packageinfo.vdf'
Mon Jan  9 07:15:21 PM MST 2023 INFO - setSteamPath - Set 'STEAMCOMPATOOLS' to '/home/shop/.local/share/Steam/compatibilitytools.d'
Mon Jan  9 07:15:21 PM MST 2023 INFO - setSteamPath - Set 'ICODIR' to '/home/shop/.local/share/Steam/steam/games'
Mon Jan  9 07:15:21 PM MST 2023 INFO - setSteamPaths - Found SteamUserId '103117483'
Mon Jan  9 07:15:21 PM MST 2023 INFO - getGameOS - Starting game OS detection
Mon Jan  9 07:15:21 PM MST 2023 INFO - getGameOS - STEAM_COMPAT_DATA_PATH is not defined, so this is either a Linux Game or no game was started at all
Mon Jan  9 07:15:21 PM MST 2023 INFO - createDefaultCfgs - createProtonList:
Mon Jan  9 07:15:21 PM MST 2023 INFO - createProtonList - Looking for available Proton versions
Mon Jan  9 07:15:21 PM MST 2023 INFO - getAvailableProtonVersions - Initially creating an array with available Proton versions
Mon Jan  9 07:15:21 PM MST 2023 INFO - getAvailableProtonVersions - Adding Proton versions found in STEAMCOMPATOOLS '/home/shop/.local/share/Steam/compatibilitytools.d'
Mon Jan  9 07:15:21 PM MST 2023 INFO - getAvailableProtonVersions - Found proton directory: '/home/shop/.local/share/Steam/compatibilitytools.d/GE-Proton7-18/proton'
Mon Jan  9 07:15:21 PM MST 2023 INFO - getAvailableProtonVersions - Found proton directory: '/home/shop/.local/share/Steam/compatibilitytools.d/GE-Proton7-20/proton'
Mon Jan  9 07:15:21 PM MST 2023 INFO - getAvailableProtonVersions - Found proton directory: '/home/shop/.local/share/Steam/compatibilitytools.d/Proton-6.5-GE-1/proton'
Mon Jan  9 07:15:21 PM MST 2023 INFO - getAvailableProtonVersions - Found proton directory: '/home/shop/.local/share/Steam/compatibilitytools.d/GE-Proton7-41/proton'
Mon Jan  9 07:15:21 PM MST 2023 INFO - getAvailableProtonVersions - Found proton directory: '/home/shop/.local/share/Steam/compatibilitytools.d/GE-Proton7-17/proton'
Mon Jan  9 07:15:22 PM MST 2023 INFO - getAvailableProtonVersions - Found proton directory: '/home/shop/.local/share/Steam/compatibilitytools.d/GE-Proton7-10/proton'
Mon Jan  9 07:15:22 PM MST 2023 SKIP - getAvailableProtonVersions - Directory SYSSTEAMCOMPATOOLS '/usr/share/steam/compatibilitytools.d' not found - skipping
Mon Jan  9 07:15:22 PM MST 2023 INFO - getAvailableProtonVersions - Adding Proton versions found in additional SteamLibrary Paths
Mon Jan  9 07:15:22 PM MST 2023 INFO - getAvailableProtonVersions - Found proton directory: '/mnt/Linux2/SteamGames/steamapps/common/Proton 5.13'
Mon Jan  9 07:15:22 PM MST 2023 INFO - getAvailableProtonVersions - Found proton directory: '/mnt/Linux2/SteamGames/steamapps/common/Proton 6.3'
Mon Jan  9 07:15:22 PM MST 2023 INFO - getAvailableProtonVersions - Found proton directory: '/mnt/Linux2/SteamGames/steamapps/common/Proton 7.0'
Mon Jan  9 07:15:22 PM MST 2023 INFO - getAvailableProtonVersions - Adding Proton versions found in CUSTOMPROTONLIST '/home/shop/.config/steamtinkerlaunch/protonlist.txt'
Mon Jan  9 07:15:22 PM MST 2023 INFO - getAvailableProtonVersions - Checking line '/home/shop/.config/steamtinkerlaunch/proton/custom/GE-Proton7-24/proton' in '/home/shop/.config/steamtinkerlaunch/protonlist.txt'
Mon Jan  9 07:15:22 PM MST 2023 INFO - getAvailableProtonVersions - File '/home/shop/.config/steamtinkerlaunch/proton/custom/GE-Proton7-24/proton' exists - adding it to the list
Mon Jan  9 07:15:22 PM MST 2023 INFO - createDefaultCfgs - createProtonList end
Mon Jan  9 07:15:22 PM MST 2023 INFO - listAllSettingsEntries - Creating '/dev/shm/steamtinkerlaunch/setentries.txt'
Mon Jan  9 07:15:24 PM MST 2023 INFO - createDefaultCfgs - STOP
Mon Jan  9 07:15:24 PM MST 2023 INFO - listSteamLibraries - (Re-)creating '/dev/shm/steamtinkerlaunch/SteamLibraries.txt'
Mon Jan  9 07:15:24 PM MST 2023 INFO - setSteamLibraryPaths - STEAM_COMPAT_LIBRARY_PATHS set to '/home/shop/.local/share/Steam/steamapps:/mnt/Linux2/SteamGames/steamapps'
Mon Jan  9 07:15:24 PM MST 2023 INFO - setSteamLibraryPaths - STEAM_COMPAT_MOUNTS set to '/home/shop/.local/share/Steam/steamapps/common/Steamworks Shared:/mnt/Linux2/SteamGames/steamapps/common/SteamLinuxRuntime_soldier'
Mon Jan  9 07:15:24 PM MST 2023 INFO - setSteamLibraryPaths - STEAM_COMPAT_TOOL_PATHS set to '/home/shop/.local/share/Steam/steamapps/common/Steamworks Shared:/mnt/Linux2/SteamGames/steamapps/common/SteamLinuxRuntime_soldier'
Mon Jan  9 07:15:24 PM MST 2023 INFO - main - Checking internal dependencies:
Mon Jan  9 07:15:24 PM MST 2023 INFO - checkIntDeps - Result of version check for yad binary '/usr/bin/yad' is '12.3'
Mon Jan  9 07:15:24 PM MST 2023 INFO - main - Initializing first Proton:
Mon Jan  9 07:15:24 PM MST 2023 INFO - initOldProtonArr - Creating an initial array with available Proton versions using the file '/dev/shm/steamtinkerlaunch/ProtonCSV.txt' which was created during a previous run
Mon Jan  9 07:15:24 PM MST 2023 INFO - initFirstProton - Initializing Proton
Mon Jan  9 07:15:24 PM MST 2023 INFO - initFirstProton - Initial Proton version 'FIRSTUSEPROTON' from '/home/shop/.config/steamtinkerlaunch/default_template.conf' is 'proton-7.0-5'
Mon Jan  9 07:15:24 PM MST 2023 INFO - main - Initializing default window resolution
Mon Jan  9 07:15:24 PM MST 2023 INFO - getScreenRes - Found screen width '1920'
Mon Jan  9 07:15:24 PM MST 2023 INFO - getScreenRes - Found screen height '1080'
Mon Jan  9 07:15:24 PM MST 2023 INFO - setInitWinXY - Using '1440x810' as default resolution for all windows without a configured resolution
Mon Jan  9 07:15:24 PM MST 2023 INFO - main - Welcome to SteamTinkerLaunch
Mon Jan  9 07:15:24 PM MST 2023 INFO - main - Checking command line: incoming arguments '--version'
Mon Jan  9 07:15:24 PM MST 2023 INFO - stlCleanup - Ensuring tray icon is definitely removed
Mon Jan  9 07:15:24 PM MST 2023 INFO - stlCleanup - Closing SteamTinkerLaunch

**Comment by tetoNidan** [comment]
Just tried to go back to the last point when MO2 launched, It was with steam beta. I redownloaded steam beta and tested steamtinkerlaunch-git and steamtinkerlaunch from the AUR and get the same output.

**Comment by sonic2kk** [comment]
The log you pasted into your initial issue seems to be for when you checked the SteamTinkerLaunch version. When you're able, please attach a log from a run of SteamTinkerLaunch where MO2 crashes.

The permissions error you showed sounds like a general system problem, try modifying the permissions of that file and seeing if you get any further.

Because of the log issue I can't see what version of Proton you're trying to use with MO2 but in the past users reported issues with GE-Proton7-43, I would try GE-Proton7-42 (one user said they had to reinstall, but in my brief tests in the past I didn't have to).

Nothing has changed on the STL side with MO2 recently and it looks like nothing has changed for MO2 either (last release was still Dec 2021).

Hopefully the Proton version change resolved this problem. You can change the Proton version in the Game Menu and Global Menu. Good luck!

**Comment by sonic2kk** [comment]
I did some research for #716, which is a very similar issue to this.

The tl;dr is that several Proton versions aren't working for me even though they used to, and they now produce a Wine error of `wine: could not load kernel32.dll, status <blah>`. I'm not sure what changed, I would *wager* a system dependency broke this as the MO2 code hasn't been touched, older versions of Proton exhibit this behaviour which used to work, and MO2 itself hasn't been updated (their last commit was almost a year ago).

I managed to get MO2 working with Proton 6.3-8, which is much less than ideal. I have no idea what broke this, but I don't think it should be anything on the SteamTinkerLaunch side. The specific status code error I'm seeing from Wine is `c0000135`, which is the exact same version reported in an old Proton issue dating back to Proton 5.0: https://github.com/ValveSoftware/Proton/issues/4269

EDIT: For vanilla Wine I also found a thread on the Manjaro forums, it suggests that this was a prefix issue which is not good news at all. But then I wonder why Proton 6 is working on my end at least. Very strange...

This seems to be some kind of Steam client issue maybe? I am not totally sure. I only skimmed the Proton issue. Games seem to be working fine on my system though.

<hr>

I would be interested to know if you're also seeing the `kernel32.dll` error message, when you are able to reply.

Trying to run anything from the MO2 prefix fails for me, even `winecfg`. Could be that the prefix is just bad, which is very unfortunate for mods... Some prefix juggling might mitigate it though e.g. backup the old prefix, create a new prefix and then replace all the MO2 installation stuff in the new prefix with the stuff from the old prefix.

**Comment by sonic2kk** [comment]
Can confirm that a fresh prefix fixed the issue on my end. The MO2 prefix is located at `~/.config/steamtinkerlaunch/mo2/compatdata`. Change the MO2 Proton version to something like GE-Proton7-42 or maybe even better, Proton 7.0-5. You can do this from the Global Menu.

Then, you should rename the existing `compatdata` at `~/.config/steamtinkerlaunch/mo2` (something like `borked_compatdata` will do) and then run `steamtinkerlaunch mo2 start` to reinstall MO2. Afterwards, MO2 should open. Follow the initial MO2 steps to select an instance and then close MO2.

If you don't have any mods or don't care about the mods you had previously, you should be able to stop here (assuming you're running into the same issue I had).

Otherwise if you do have mods, open both prefixes (the new `compatdata` and the `borked_compatdata`, or whatever you called yours) and relative to the `compatdata` folder, navigate to `pfx/drive_c/users/steamuser/AppData/Local`. From here, you'll want to do two things:
1. From the ***new*** compatdata folder, remove the `ModOrganizer` and `LOOT` folder. (or move, if you have disk space to spare - In my case I don't have enough disk space)
2. From the **old** compatdata folder (`borked_compatdata`) copy over your `ModOrganizer` and `LOOT` folder.

This should copy over your mods and other MO2 instance data. It won't keep any MO2 themes or anything but you could probably copy those over too, or copy over all your MO2 installation files. I didn't go that far personally but if you have MO2 tweaked the way you like you could copy the configuration files, I believe they're stored in the MO2 installation folder which should be at `pfx/drive_c/Modding/MO2` (in the *old* prefix).

<hr>

If you're having a different issue then this may all be for naught, but that's been the extent of my research. It seems like trying to run MO2 with GE-Proton7-43 completely borks the prefix for some reason.

**Comment by sonic2kk** [comment]
For the record, #716 is a separate issue relating to `innoextract` on Steam Deck not updating.

**Comment by tetoNidan** [comment]
Is there a reason a new STL.log is not being generated? I ran STL a few times since I did the --version check and still have the same log from:

Mon Jan  9 07:15:24 PM MST 2023 INFO - main - Checking command line: incoming arguments '--version'

> I'm not sure what changed, I would wager a system dependency broke this as the MO2 code hasn't been touched

I don't think it's a steam runtime issue or system dependency issue because I had MO2 working when I initially switched to bata yesterday using steamtinkerlaunch-v11.12.20221020-3 and I haven't updated anything on my system in about a month. But I could be wrong as I don't know the inner working of STL at all.

> I would be interested to know if you're also seeing the kernel32.dll error message, when you are able to reply.

I did not see the kerrnel32.dll message, there was tons of output so it might have been at the beginning and I didn't catch it. How would I check wine error code for c0000135? Might be easier than trying to find the kernel32.dll in the output or I guess I could redirect steams output to a text file like so: steam > ~/Desktio/steam.log : and search for the kernel32.dll message?

I will look through the output again for kernel32.dll and find the file that is in read only mode and give it permissions for writing as well as try existing glorious proton versions  I have installed and download older ones if need be, If that doesn't work I will back up MO2 prefix and reinstall MO2 and move existing folders to keep my 199 mods as per your instructions. Sounds simple enough! I'll keep you posted, might be this weekend sometime before I can really dig in to it though.

Thanks [sonic2kk](https://github.com/sonic2kk) for your work on STL and rapid responses for things I should do including a wonderfully detailed instructions on fixing a borked MO2 prefix and keeping all the modding work I have intact!

**Comment by tetoNidan** [comment]
Just tried my idea of redirecting the output of: steam &> ~/Desktop/steam.log : could not find kernel32.dll in the output. Maybe that is not the best way to search for what you were looking for. Is there a better way?

Just tried to find ModOrganizer-steamtinkerlaunch-dl.desktop-old.

ls -la /home/shop/.local/share/applications/ModOrganizer-steamtinkerlaunch-dl.desktop-old
s: cannot access '/home/shop/.local/share/applications/ModOrganizer-steamtinkerlaunch-dl.desktop-old': No such file or directory.

I just reread the output and I guess I should have looked for the .desktop file not the .desktop-old that it was trying to move it to. I was looking at permissions and all the .desktop files have the same permissions of -rw-r--r-- "0644"

is this the correct permissions for applications folder? Shouldn't it be more like drwxr-xr-x? Looks like its 0700 not 0755? Could that be the issue?

ls -la /home/shop/.local/share/ | grep applications
drwx------  2 shop shop  4096 Nov 30 14:03 applications

I'm not exactly a linux noob but defiantly not a guru either! Not sure if the permissions are off for some reason. I dont personally go changing them and don't know how they got changed if these are not the right permissions?

Thanks again [sonic2kk](https://github.com/sonic2kk)!

**Comment by sonic2kk** [comment]
> Is there a reason a new STL.log is not being generated? I ran STL a few times since I did the --version check and still have the same log 

That's a little strange. The new log should be generated at `/dev/shm/steamtinkerlaunch`. Try deleting that folder and running STL again, STL should always generate a new log.

> I don't think it's a steam runtime issue or system dependency issue because I had MO2 working when I initially switched to bata yesterday using steamtinkerlaunch-v11.12.20221020-3 and I haven't updated anything on my system in about a month. But I could be wrong as I don't know the inner working of STL at all.

Yeah, I think maybe this is a prefix issue. If MO2 runs once with GE-Proton7-43, the prefix seems to break. I tested this myself a couple of times. But we'll get to that in a sec :-)

> I did not see the kerrnel32.dll message, there was tons of output so it might have been at the beginning and I didn't catch it

You seem to be running Steam from the command line and looking at the output, which is not ideal. It may not even show up from the Steam log. Replying a bit out of sequence here but this relates to another point you asked about:

> Maybe that is not the best way to search for what you were looking for. Is there a better way?

Yeah, you should be able to see the logging output if you run `steamtinkerlaunch mo2 start`, it will display the Wine output logging. The logging you're looking at is from Steam, which is separate from SteamTinkerLaunch.

> [...]  including a wonderfully detailed instructions on fixing a borked MO2 prefix and keeping all the modding work I have intact!

Thank you :-) I even documented it on the wiki in case this happens again (Wine prefixes can bork occasionally, had it happen a few times over the years)

> Sounds simple enough! I'll keep you posted, might be this weekend sometime before I can really dig in to it though.

Heh, I don't have quite as many mods as that but I definitely started sweating when I tested and saw MO2 wouldn't boot for me either. It can be a delicate and stressful process, at the very worst symlinking the folder from the old prefix to the new prefix could work, if you're worried about moving files or if it might take too long. I don't expect anything to go wrong, this is more of a friendly "be careful" from one modder to another :sweat_smile: 

<hr>

> I was looking at permissions and all the .desktop files have the same permissions of -rw-r--r-- "0644". is this the correct permissions for applications folder? Shouldn't it be more like drwxr-xr-x? Looks like its 0700 not 0755? Could that be the issue?

I was initially hopeful for this but it seems like my permissions are the same. Not sure why there's a permissions issue. You could try deleting the MO2 `.desktop` files and re-creating them, maybe something got messed up somehow. Though I think this error is not a *huge* problem for now, it shouldn't be preventing MO2 from launching as that is handled by Wine - The `.desktop` files here iirc are for the link handling only. The actual launching of MO2 is handled by the Bash code which just runs a command to launch it with a specific Wine version in a specific prefix. Actually when MO2 boots properly you can see the command SteamTinkerLaunch uses to run MO2, in my case it is: `WINEDEBUG="-all" WINEPREFIX="/home/user/.config/steamtinkerlaunch/mo2/compatdata/pfx" "/home/user/.local/share/Steam/compatibilitytools.d/GE-Proton7-42/files/bin/wine" "/home/user/.config/steamtinkerlaunch/mo2/compatdata/pfx/drive_c/Modding/MO2/ModOrganizer.exe"` - I know you said you're not a Linux guru but maybe it's a helpful technical detail :-) You could run the MO2 manually this way, just replacing the paths with your paths. For example you might have a different path to Wine. Or if you're lucky you might only have to change the username to run it manually!

We can come back to the `.desktop` file permission error though if removing the files doesn't help and once we get MO2 actually booting up. I understand and agree with investigating and fixing it but it's probably more important to get MO2 booting for now :-)

> I'm not exactly a linux noob but defiantly not a guru either! Not sure if the permissions are off for some reason. I dont personally go changing them and don't know how they got changed if these are not the right permissions?

I'm wondering how they got changed as well. I definitely don't think you went out of your way to change them, and SteamTinkerLaunch shouldn't have broken it but maybe something went wrong. Hopefully removing the file and replacing it fixes it (not sure if you already tried that), as SteamTinkerLaunch should re-generate it.

Good luck!

**Comment by tetoNidan** [comment]
So I have it working! Though I'm pretty sure its a steam issue, or that's my best guess at the moment. Here is my adventure in troubleshooting.

I followed your steps to fix a borked MO2 prefix. tried a few older GE versions. The GE version I was using when things borked was 7-41.  I tried 7-18 and 7-10 as I already had them downloaded. Each time removing the old MO2 folder, reinstalling MO2 and replacing the folders as advised. No luck each time. 

I should not that each time I tried a new proton I was able to start steamtinkerlaunch mo2 start. I was meaning to try that before I started switching to different protons but my adhd got the better of me and I did not. For all I know MO2 might have started just fine. That will be my first thing I try if or when I have any issues in the future. I was unaware STL would start a standalone instance of MO2. Any way...

Downloaded GE_7-42 and did the removal, install and replace but still MO2 would not start. That time I had started steam through the terminal to watch the output and I caught something about steam having trouble killing something that didnt believe steam was its parent. 

This got me thinking it was something inside steam not killing something properly. Maybe something in a cache or ram, who knows? I did a reboot and the reboot's output wasn't as clean as it usually is. I'm suing LUKS and am used to seeing some output about cleaning blocks, usually one line. This time the screen flashed after it says cleaning blocks with more output to fast to really get, flashed again and then rebooted like normal.

I fired up steam normally and MO2 failed again. So I I removed the .desktop file from applications folder and thought I'd capture the output so I redirected it to a text file started in the terminal and MO2 started just fine! 

I'm don't know for sure what or exactly why but i'm thinking a steam or wine dependency was not shutting down and steam was not able to control it causing the issue. Maybe it was that .desktop file was corrupted somehow?

I should have more free time this weekend and I'll try to bork the game on purpose by installing the steam beta again. I want to see if I cant get the controller working with firejail in the new steam overlay UI anyway. If I manage to bork and fix maybe I can get a better idea of whats going on? The steps you gave to fix the borked MO2 are super simple and I'm pretty confident if I break it again it will only be a couple hours max before I get things running again!

I know this last post is absurdly unhelpful for troubleshooting as all I have is wild guesses as to what actually broke it and fixed it!  

Two hours before midnight and I think I just might go kill Miraak! 

Thanks once again [sonic2kk](https://github.com/sonic2kk) for all you do maintaining STL! Like I said I'll try to break it on purpose this sunday or monday, if you want to keep this open I will post the details hopefully with something slightly better than hay it works now, don't know why?

**Comment by sonic2kk** [comment]
Thanks for the super investigation here! I'm glad you've got MO2 working now!

There could be something going on here with Steam or Wine processes lingering. I run Amazon Music via Wine and every so often if I close it the WIneserver won't properly shut down, and when I try to open it again it won't work. I haven't used LUKS before so I'm not sure if it could be causing any problems with STL, I hope not though! :-)

> I was unaware STL would start a standalone instance of MO2

Ah yeah, I should've been a bit clearer here. `steamtinkerlaunch mo2 start` will start a standalone MO2, this is basically the same thing as using the MO2 Extra Button on the Main Menu. This runs using its own version of Proton which you can select from the Global Menu. It also runs MO2 in its own prefix at `~/.config/steamtinkerlaunch/mo2/compatdata`. But when you run MO2 in "Game Mode" (when you get the dialog prompt after pressing the STL "Play" button) this will use the game's Proton version and the game's prefix.

The separate Proton versions are, if I recall correctly, mainly for legacy reasons. Previously MO2 relied on `dotnet48` which we had a [custom version of Proton](https://github.com/frostworx/steamtinkerlaunch-tweaks/releases/tag/Proton-6.17-STL-1) to install. Somewhere along the way MO2 updated to no longer require `dotnet48` and was fully compatible with Wine's implementation of .NET called Mono. Basically MO2 used to need a custom version of Proton to install properly, and MO2 is installed when you use the MO2 Extra Button or the `steamtinkerlaunch mo2 start` command. It isn't needed anymore though, but it would be a bit of effort to remove it, and some users may find it useful (maybe a specific Proton version is useful for installing a given mod or running a given MO2 mod program in standalone mode, but the game you want to play won't work with that custom Proton version).

> I fired up steam normally and MO2 failed again. So I I removed the .desktop file from applications folder and thought I'd capture the output so I redirected it to a text file started in the terminal and MO2 started just fine!

This is also quite interesting. I'll take another look at the MO2 stuff when I have a free moment but I didn't think that this could be causing problems. I suppose something that could be happening here is that as you said, the MO2 dekstop file got corrupted somehow, and when STL tried to read it then something got messed up.

> I should have more free time this weekend and I'll try to bork the game on purpose by installing the steam beta again. I want to see if I cant get the controller working with firejail in the new steam overlay UI anyway

Sounds good! I'll keep this issue open until then, even though it is "solved" for now, finding out what might've went wrong is useful in case any changes are needed on the STL side, and also in case there is something here which could be documented on the wiki.

> Two hours before midnight and I think I just might go kill Miraak!

Hehe, I assume you didn't lose any mods then :-) Phew!

<hr>

Definitely happy to keep this issue open for any further testing, I would also be interested to know what caused the problem. Given that you were using GE-Proton7-42 before I would guess the problem isn't related to GE-Proton7-43 like I thought, but perhaps something is going wrong with the MO2 .desktop file.

Once I have a free moment I can attempt to mess up my MO2 .desktop file and see if that causes problems. I was fairly sure that the file was only used when the Nexusmods Mod Manager button was pressed but I think STL does try to set up some downloader stuff when MO2 is opened, maybe it tries to read/copy the contents of this file and fails, and that causes MO2 to not open for some reason.

Thanks a bunch :smile:

**Comment by tetoNidan** [comment]
So I tried to reproduce what had happened and I had no luck. I can say that it would be very unlikely anything to do with LUKS though as it just unlocks encryption. I only mentioned it because it acted different on shutdown possibly because of a program not responding to termination. Might have been the same one steam was complaining about?

I'm at a loss for explanation or further insight as to what happened. 

As always thanks allot. Love me some STL!

**Comment by sonic2kk** [comment]
Glad you didn't run into it again at least so hopefully it isn't a major issue and was just a little "blip". I'll keep it in mind though if any other users run into this problem as it seems like a straightforward fix at least!

Thanks for your research into this and help with debugging, it's also much appreciated :-)
